theme/views.py

```diff
@@ -156,3 +168,122 @@ def signup(request, template="accounts/account_signup.html"):
         "title": _("Sign up"),
     }
     return render(request, template, context)
+
+
+@login_required
+def update_user_profile(request):
+    user = request.user
+    old_email = user.email
+    user_form = UserForm(request.POST, instance=user)
+    user_profile = UserProfile.objects.filter(user=user).first()
+    profile_form = UserProfileForm(request.POST, request.FILES, instance=user_profile)
+    try:
+        with transaction.atomic():
+            if user_form.is_valid() and profile_form.is_valid():
+                user_form.save()
+
+                password1 = request.POST['password1']
+                password2 = request.POST['password2']
+                if len(password1) > 0:
+                    if password1 == password2:
+                        user.set_password(password1)
+                        user.save()
+                    else:
+                        raise ValidationError("Passwords do not match.")
+
+                profile = profile_form.save(commit=False)
+                profile.user = request.user
+                profile.save()
+                messages.success(request, "Your profile has been successfully updated.")
+                # if email was updated, reset to old email and send confirmation
+                # email to the user's new email - email will be updated upon confirmation
+                if old_email != user.email:
+                    new_email = user.email
+                    user.email = old_email
+                    user.save()
+                    # send a confirmation email to the new email address
+                    send_verification_mail_for_email_update(request, user, new_email, "email_verify")
+                    info(request, _("A verification email has been sent to your new email with "
+                                    "a link for updating your email. If you "
+                                    "do not receive this email please check your "
+                                    "spam folder as sometimes the confirmation email "
+                                    "gets flagged as spam. If you entered an incorrect "
+                                    "email address, please request email update again. "
+                                    ))
+                    # send an email to the old address notifying the email change
+                    send_mail(subject="Your HydroShare email.",
```

Comment:
Change subject to "A request to change this email address in HydroShare has been made" or something along that line?